docs/designers-developers/developers/tutorials/format-api/2-toolbar-button.md

```diff
@@ -35,3 +35,49 @@ Let's check that everything is working as expected. Reload the post/page and sel
 ![Toolbar with custom button](https://raw.githubusercontent.com/WordPress/gutenberg/master/docs/designers-developers/assets/toolbar-with-custom-button.png)
 
 You may also want to check that upon clicking the button the `toggle format` message is shown in your browser's console.
+
+By default, the button is rendered on every rich text toolbar (image captions, buttons, paragraphs, etc).
+It is possible to render the button only on blocks of a certain type by using `wp.data.withSelect` together with `wp.compose.ifCondition`.
+The following sample code renders the previously shown button only on paragraph blocks:
+
+```js
+( function( wp ) {
+	var withSelect = wp.data.withSelect;
+	var ifCondition = wp.compose.ifCondition;
+	var compose = wp.compose.compose;
+	var MyCustomButton = function( props ) {
+		return wp.element.createElement(
+			wp.editor.RichTextToolbarButton, {
+				icon: 'editor-code',
+				title: 'Sample output',
+				onClick: function() {
+					console.log( 'toggle format' );
+				},
+			}
+		);
+	}
+	var ConditionalButton = compose(
+		withSelect( function( select ) {
+			return {
+				selectedBlock: select('core/editor').getSelectedBlock()
+			}
+		} ),
+		ifCondition( function( props ) {
+			return (
+				props.selectedBlock &&
+				props.selectedBlock.name === 'core/paragraph'
+			); 
+		} )
+	)( MyCustomButton );
```

Comment:
Nitpick: let's introduce a line break here as a sort of breadth between the components declaration and the format registration.